src/Symfony/Component/Messenger/Handler/Locator/ContainerHandlerLocator.php

```diff
@@ -20,15 +20,17 @@
 class ContainerHandlerLocator extends AbstractHandlerLocator
 {
     private $container;
+    private $handlerKeyFormat;
 
-    public function __construct(ContainerInterface $container)
+    public function __construct(ContainerInterface $container, string $handlerKeyFormat = 'handler.%s')
     {
         $this->container = $container;
+        $this->handlerKeyFormat = $handlerKeyFormat;
     }
 
     protected function getHandler(string $class)
     {
-        $handlerKey = 'handler.'.$class;
+        $handlerKey = sprintf($this->handlerKeyFormat, $class);
```

Comment:
why do we need a format at all? shouldn't the locator be built with the class as the key instead directly?

Reply:
The prefix was mainly advocated by @dunglas (and you if I remember correctly 😄) to be sure that we don't misinterpret what the key is about (i.e. the message, not the handler).

Reply:
Any link?